backend/app/adapter/sqldb/userrole_integration_test.go

```diff
@@ -0,0 +1,227 @@
+// +build integration all
+
+package sqldb_test
+
+import (
+	"database/sql"
+	"fmt"
+	"testing"
+
+	"github.com/short-d/app/fw/assert"
+	"github.com/short-d/app/fw/db/dbtest"
+	"github.com/short-d/short/backend/app/adapter/sqldb"
+	"github.com/short-d/short/backend/app/adapter/sqldb/table"
+	"github.com/short-d/short/backend/app/entity"
+	"github.com/short-d/short/backend/app/usecase/authorizer/role"
+)
+
+var insertUserRoleRowSQL = fmt.Sprintf(`
+INSERT INTO "%s" ("%s", "%s")
+VALUES ($1, $2)`,
+	table.UserRole.TableName,
+	table.UserRole.ColumnUserID,
+	table.UserRole.ColumnRole,
+)
+
+type userRoleTableRow struct {
+	userID string
+	role   role.Role
+}
+
+func TestUserRoleSQL_AddRole(t *testing.T) {
+	testCases := []struct {
+		name          string
+		user          entity.User
+		userRoleRows  []userRoleTableRow
+		newRoles      []role.Role
+		expectedRoles []role.Role
+		hasErr        bool
+	}{
+		{
+			name: "add 1 role for nonexistent user",
+			user: entity.User{
+				ID: "1343",
+			},
+			userRoleRows: []userRoleTableRow{
+				{"4444", role.Basic},
+			},
+			newRoles:      []role.Role{role.ChangeLogViewer},
+			expectedRoles: []role.Role{role.ChangeLogViewer},
+			hasErr:        false,
```

Comment:
shouldn't hasErr be true?

Reply:
We didn't have foreign key constraint yet. Will fix after we migrated from email to user ID.